ggml-qnn.cpp

```diff
@@ -0,0 +1,3590 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <stddef.h>
+#include <inttypes.h>
+#include <math.h>
+#include <time.h>
+#include <unistd.h>
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+
+#include <string>
+#include <vector>
+#include <thread>
+#include <mutex>
+#include <map>
+#include <set>
+#include <tuple>
+#include <queue>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <chrono>
+#include <memory>
+#include <regex>
+#include <random>
+#include <functional>
+#include <unordered_map>
+#include <condition_variable>
+#include <cassert>
+#include <unordered_set>
+#include <utility>
+#include <stdatomic.h>
+
+#include "QnnTypes.h"
+#include "QnnCommon.h"
+#include "QnnContext.h"
+#include "QnnBackend.h"
+#include "QnnGraph.h"
+#include "QnnProperty.h"
+#include "QnnTensor.h"
+#include "QnnInterface.h"
+#include "Saver/QnnSaver.h"
+#include "System/QnnSystemInterface.h"
+#include "HTP/QnnHtpDevice.h"
+
+#include "ggml-qnn.h"
+
+#include "ggml-backend-impl.h"
+
+
+// =================================================================================================
+//
+//  forward/external/helper declaration
+//
+// =================================================================================================
+class qnn_instance;
+
+
+#if (defined __ANDROID__) || (defined ANDROID)
+extern "C" int __android_log_print(int prio, const char * tag, const char * fmt, ...)
```

Comment:
nit: better to remove the extern function here and then include the ndk's `log.h` instead, to prevent the protential change of this interface.

Reply:
> nit: better to remove the extern function here and then include the ndk's `log.h` instead, to prevent the protential change of this interface.

thanks.I'll modify it accordingly.